test/core/gprpp/inlined_vector_test.cc

```diff
@@ -87,14 +87,14 @@ TEST(InlinedVectorTest, ClearAndRepopulate) {
 }
 
 TEST(InlinedVectorTest, ConstIndexOperator) {
-  const int kNumElements = 10;
+  constexpr int kNumElements = 10;
   InlinedVector<int, 5> v;
   EXPECT_EQ(0UL, v.size());
   for (int i = 0; i < kNumElements; ++i) {
     v.push_back(i);
     EXPECT_EQ(i + 1UL, v.size());
   }
-  auto const_func = [kNumElements](const InlinedVector<int, 5>& v) {
+  auto const_func = [](const InlinedVector<int, 5>& v) {
```

Comment:
Ugh, @vjpai , turns out CL (windows) doesn't like this, even though this is perfectly standard complaint (see [here](https://stackoverflow.com/questions/42610429/must-constexpr-expressions-be-captured-by-a-lambda-in-c)). We can make everyone happy if I write `[=]` for the capture group, but then you won't be happy because anonymous captures :/

Reply:
Here's a quick way to play with it: https://godbolt.org/g/cepeMR You can choose clang 5.0 as well, adding the flags `-std=c++11 -Wunused-lambda-capture -Werror`

Reply:
Something gotta give, because either clang 5 will choke, or MS CL will choke.

Reply:
FWIW, my read of the style guide is essentially the same as Vijay's: we should avoid default captures when at all possible.

In this case, it may not be possible due to the Windows compiler being dumb.  If so, I think it's fine to leave it here (especially since this is a test, not production code) with a comment explaining why it's necessary.

Reply:
Done. @vjpai PTAL.